File: ClimbEdge.Application/DTOs/ClimbZoneDTO.cs
namespace ClimbEdge.Application.DTOs
{
    public record class CreateClimbZoneDTO
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public bool IsPublic { get; set; } = true;
        public bool IsIndoor { get; set; } = false;
        public string? ImageUrl { get; set; }
        public long? OrganizationId { get; set; }
        /// <summary>Optional WKT string (POINT Z) for the zone's GPS location.</summary>
        public string? LocationWkt { get; set; }
    }

    public record class UpdateClimbZoneDTO
    {
        public string? Name { get; set; }
        public string? Description { get; set; }
        public bool? IsPublic { get; set; }
        public string? ImageUrl { get; set; }
        public long? OrganizationId { get; set; }
        /// <summary>Optional WKT string (POINT Z) to update the GPS location.</summary>
        public string? LocationWkt { get; set; }
    }

    public record class GetClimbZoneDTO : BaseDTO
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public bool IsPublic { get; set; }
        public bool IsIndoor { get; set; }
        public string? ImageUrl { get; set; }
        public long? OrganizationId { get; set; }
        /// <summary>Latitude of the zone, if available.</summary>
        public double? Latitude { get; set; }
        /// <summary>Longitude of the zone, if available.</summary>
        public double? Longitude { get; set; }
    }
}

File: ClimbEdge.App/shared/src/types/ClimbZoneDTO.ts
import { BaseDTO } from './BaseDTO';

export interface CreateClimbZoneDTO {
  name: string;
  description?: string;
  isPublic: boolean;
  isIndoor: boolean;
  imageUrl?: string | null;
  organizationId?: number | null;
}

export interface UpdateClimbZoneDTO {
  name?: string;
  description?: string;
  isPublic?: boolean;
  imageUrl?: string | null;
  organizationId?: number | null;
}

export class GetClimbZoneDTO extends BaseDTO {
  name!: string;
  description?: string;
  isPublic!: boolean;
  isIndoor!: boolean;
  imageUrl?: string | null;
  organizationId?: number | null;
}
